velox/functions/lib/aggregates/DecimalAggregate.h

```diff
@@ -71,7 +71,26 @@ struct LongDecimalWithOverflowState {
 template <typename TResultType, typename TInputType = TResultType>
 class DecimalAggregate : public exec::Aggregate {
  public:
-  explicit DecimalAggregate(TypePtr resultType) : exec::Aggregate(resultType) {}
+  explicit DecimalAggregate(TypePtr inputType, TypePtr resultType)
+      : exec::Aggregate(resultType) {
+    // If input's precision is much less than 38, we can skip the overflow check
+    // for first skipOverflowCheckCount_ rows. For exmaple, Decimal(7, 2), the
+    // max value is 99999.99, the overflow happens when there are more than
+    // pow(10, 31) rows.
+    if (inputType->isShortDecimal()) {
+      auto precision = 38 - inputType->asShortDecimal().precision();
+      if (precision > 19) {
+        skipOverflowCheckCount_ = std::numeric_limits<uint64_t>::max();
```

Comment:
The `count` data type is int64_t, should it be std::numeric_limits<int64_t>::max()